homeassistant/components/template/helpers.py

```diff
@@ -0,0 +1,40 @@
+"""Helpers for template integration."""
+
+import logging
+
+from homeassistant.components import blueprint
+from homeassistant.const import SERVICE_RELOAD
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.singleton import singleton
+
+from .const import DOMAIN, TEMPLATE_BLUEPRINT_SCHEMA
+
+DATA_BLUEPRINTS = "template_blueprints"
+
+LOGGER = logging.getLogger(__name__)
+
+
+def _blueprint_in_use(hass: HomeAssistant, blueprint_path: str) -> bool:
+    """Return True if any template references the blueprint."""
+    from . import templates_with_blueprint  # pylint: disable=import-outside-toplevel
+
+    return len(templates_with_blueprint(hass, blueprint_path)) > 0
+
+
+async def _reload_blueprint_templates(hass: HomeAssistant, blueprint_path: str) -> None:
+    """Reload all templates that rely on a specific blueprint."""
+    await hass.services.async_call(DOMAIN, SERVICE_RELOAD)
```

Comment:
This line is not covered with tests as the RELOAD handler is [locally defined](https://github.com/home-assistant/core/blob/cf7205584e67c05dd04e3d718e45b5be8d27b68a/homeassistant/components/template/__init__.py#L84) in `async_setup`

Should I take it out as a separate method and patch it to test this as well?

Reply:
Since the reload helper should be tested separately, I think it's enough to test that this callback calls the service. We can do that by mocking the service with that test helper, using the same service domain and name.

https://github.com/home-assistant/core/blob/781c3eed2fa95ee8f765761e9b40f578df4cca2a/tests/common.py#L346C5-L355